--- canokey-qemu-1/CMakeLists.txt.orig	2026-03-24 02:10:51.000000000 +0100
+++ canokey-qemu-1/CMakeLists.txt	2026-05-18 20:19:22.101959275 +0200
@@ -25,7 +25,6 @@ set(LIBCANOKEY_QEMU_INCLUDEDIR ${CMAKE_I
 set(QEMU_VIRT_SRCS
     canokey-core/virt-card/device-sim.c
     canokey-core/virt-card/fabrication.c
-    canokey-core/littlefs/bd/lfs_filebd.c
 )
 file(GLOB_RECURSE SRC Src/*.c)
 
@@ -40,7 +39,7 @@ set_target_properties(canokey-qemu PROPE
 )
 target_include_directories(canokey-qemu SYSTEM PRIVATE Inc canokey-core/virt-card canokey-core)
 target_compile_definitions(canokey-qemu PRIVATE HW_VARIANT_NAME="CanoKey QEMU")
-target_link_libraries(canokey-qemu PRIVATE canokey-core)
+target_link_libraries(canokey-qemu PRIVATE canokey-core lfs)
 add_dependencies(canokey-qemu gitrev)
 
 install(TARGETS canokey-qemu
--- canokey-qemu-1/canokey-core/CMakeLists.txt.orig	2026-03-15 13:33:33.000000000 +0100
+++ canokey-qemu-1/canokey-core/CMakeLists.txt	2026-05-18 20:22:16.204349415 +0200
@@ -78,11 +78,8 @@ file(
     src/*.c
     applets/*.c
     interfaces/*.c
-    littlefs/lfs.c
-    littlefs/lfs_util.c
-    tinycbor/src/cborencoder.c
-    tinycbor/src/cborparser.c
 )
+#include_directories("/usr/include/tinycbor")
 add_library(canokey-core ${SRC})
 add_dependencies(canokey-core generate_ctap_get_info_cbor)
 
@@ -102,9 +99,7 @@ target_include_directories(
     SYSTEM
     PUBLIC
         include
-        littlefs
-        tinycbor/src
-        ${CMAKE_CURRENT_BINARY_DIR}/tinycbor
+        /usr/include/tinycbor
         ${CMAKE_CURRENT_BINARY_DIR}
         interfaces/USB/device
         interfaces/USB/core/inc
@@ -114,7 +109,7 @@ target_include_directories(
         interfaces/USB/class/webusb
 )
 target_compile_definitions(canokey-core PRIVATE CBOR_NO_FLOATING_POINT CBOR_STATIC_DEFINE)
-target_link_libraries(canokey-core canokey-crypto)
+target_link_libraries(canokey-core canokey-crypto lfs tinycbor)
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules")
 
@@ -146,17 +141,17 @@ if(ENABLE_TESTS)
         virt-card/device-sim.c
         virt-card/fabrication.c
         virt-card/fido-hid-over-udp.c
-        littlefs/bd/lfs_filebd.c
     )
     target_include_directories(
         fido-hid-over-udp
         SYSTEM
-        PRIVATE virt-card littlefs
+        PRIVATE virt-card
     )
     target_link_libraries(
         fido-hid-over-udp
         general
         canokey-core
+	lfs
         "-fsanitize=address"
     )
     add_dependencies(fido-hid-over-udp gitrev)
@@ -170,14 +165,13 @@ if(ENABLE_TESTS)
             virt-card/device-sim.c
             virt-card/ifdhandler.c
             virt-card/fabrication.c
-            littlefs/bd/lfs_filebd.c
         )
         target_include_directories(
             u2f-virt-card
             SYSTEM
-            PRIVATE virt-card ${PCSCLITE_INCLUDE_DIRS} littlefs
+            PRIVATE virt-card ${PCSCLITE_INCLUDE_DIRS}
         )
-        target_link_libraries(u2f-virt-card ${PCSCLITE_LIBRARIES} canokey-core)
+        target_link_libraries(u2f-virt-card ${PCSCLITE_LIBRARIES} canokey-core lfs)
         add_dependencies(u2f-virt-card gitrev)
     endif()
 endif(ENABLE_TESTS)
@@ -189,14 +183,13 @@ if(ENABLE_FUZZING)
         virt-card/usb-dummy.c
         virt-card/device-sim.c
         virt-card/fabrication.c
-        littlefs/bd/lfs_filebd.c
     )
     target_include_directories(
         honggfuzz-fuzzer
         SYSTEM
-        PRIVATE virt-card littlefs
+        PRIVATE virt-card
     )
-    target_link_libraries(honggfuzz-fuzzer canokey-core)
+    target_link_libraries(honggfuzz-fuzzer canokey-core lfs)
     add_dependencies(honggfuzz-fuzzer gitrev)
 
     add_executable(
@@ -206,13 +199,12 @@ if(ENABLE_FUZZING)
         virt-card/usb-dummy.c
         virt-card/device-sim.c
         virt-card/fabrication.c
-        littlefs/bd/lfs_filebd.c
     )
     target_include_directories(
         honggfuzz-debug
         SYSTEM
-        PRIVATE virt-card littlefs
+        PRIVATE virt-card
     )
-    target_link_libraries(honggfuzz-debug canokey-core)
+    target_link_libraries(honggfuzz-debug canokey-core lfs)
     add_dependencies(honggfuzz-debug gitrev)
 endif(ENABLE_FUZZING)
--- canokey-qemu-1/canokey-core/virt-card/fabrication.c.orig	2026-03-15 13:33:33.000000000 +0100
+++ canokey-qemu-1/canokey-core/virt-card/fabrication.c	2026-05-18 20:22:59.920779249 +0200
@@ -10,7 +10,7 @@
 #include <aes.h>
 #include <apdu.h>
 #include <assert.h>
-#include <bd/lfs_filebd.h>
+#include <lfs_filebd.h>
 #include <ctap.h>
 #include <fs.h>
 #include <lfs.h>
